### PR TITLE
added segment info and related unit tests

### PR DESCRIPTION
**Pull request by vincentberenz**
simple convenience to write info about a memory segment

### CODE REVIEW

File: include/shared_memory/shared_memory.hpp [hunk 1]
```diff
@@ -280,6 +292,14 @@ namespace shared_memory {
       const std::string &segment_id,
       const bool clear_upon_destruction=false);
 
+  /**
+   * @brief performs introspection on the segment
+   * and return related information. If the segment does not 
+   * exists, creates it first.
+   */
+  SegmentInfo get_segment_info(const std::string &segment_id,
+			       const bool clear_upon_destruction=false);
```
Comment (MaximilienNaveau) [inline include/shared_memory/shared_memory.hpp]: weird indent + why clear_upon_destruction?

**Comment by vincentberenz** [comment]
@MaximilienNaveau added the set_verbose function, to turn off console print on object creation.

File: src/shared_memory.cpp [hunk 2]
```diff
@@ -13,14 +13,18 @@ namespace bi=boost::interprocess;
 
 namespace shared_memory {
 
+  bool VERBOSE=true;
```
Comment (MaximilienNaveau) [inline src/shared_memory.cpp]: Should we set it to false by default?

File: include/shared_memory/shared_memory.hxx [hunk 3]
```diff
@@ -38,9 +44,10 @@ namespace shared_memory {
 						     get_.second);
     }
 
-    if (registered){
+
+    if (registered && VERBOSE){
       std::cout << "registration of " << object_id
```
Comment (MaximilienNaveau) [inline include/shared_memory/shared_memory.hxx]: Can we please write where this print comes from?

for example:
```std::cout << "shared_memory::SharedMemorySegment::get_object(): " <<```

File: include/shared_memory/shared_memory.hxx [hunk 4]
```diff
@@ -61,9 +68,9 @@ namespace shared_memory {
 
     bool registered = register_object(object_id, set_);
 
-    if (registered){
+    if (registered && VERBOSE){
       std::cout << "registration of " << object_id
-		<< " of size " << objects_[object_id].second << std:: endl;
+    		<< " of size " << objects_[object_id].second << std:: endl;
```
Comment (MaximilienNaveau) [inline include/shared_memory/shared_memory.hxx]: Idem, the name of the function or at lest the package name should be in the debug string. This helps a lot removing/debugging the verbosity

File: src/shared_memory.cpp (hunk 2 above)
Comment (vincentberenz) [reply src/shared_memory.cpp]: +1

File: include/shared_memory/shared_memory.hxx (hunk 3 above)
Comment (vincentberenz) [reply include/shared_memory/shared_memory.hxx]: +1

File: include/shared_memory/shared_memory.hxx (hunk 4 above)
Comment (vincentberenz) [reply include/shared_memory/shared_memory.hxx]: +1

**Comment by vincentberenz** [comment]
did the proposed changes and merged

**Pull request merged**